diff --git a/js/manage.js b/js/manage.js
index 15c0f94..db7505f 100644
--- a/js/manage.js
+++ b/js/manage.js
@@ -2,8 +2,8 @@ $(document).ready(function () {
     var apiClientId = 'application_manager';
     var apiScope = ["applications"];
 
-    var authorizeEndpoint = 'http://localhost/php-oauth/authorize.php';
-    var apiEndpoint = 'http://localhost/php-oauth/api.php';
+    var authorizeEndpoint = '{BASE_URL}/php-oauth/authorize.php';
+    var apiEndpoint = '{BASE_URL}/php-oauth/api.php';
 
     jso_configure({
         "application_manager": {
