diff --git a/js/grades.js b/js/grades.js
index 3f4bd67..d5d4f82 100644
--- a/js/grades.js
+++ b/js/grades.js
@@ -2,9 +2,9 @@ $(document).ready(function () {
     var apiClientId = 'html-view-grades';
     var apiScope = ["grades"];
 
-    var authorizeEndpoint   = 'http://localhost/php-oauth/authorize.php';
-    var entitlementEndpoint = 'http://localhost/php-oauth/api.php/resource_owner/entitlement';
-    var apiEndpoint         = 'http://localhost/php-oauth-grades-rs/api.php';
+    var authorizeEndpoint   = '{BASE_URL}/php-oauth/authorize.php';
+    var entitlementEndpoint = '{BASE_URL}/php-oauth/api.php/resource_owner/entitlement';
+    var apiEndpoint         = '{BASE_URL}/php-oauth-grades-rs/api.php';
 
     jso_configure({
         "html-view-grades": {
