diff -urN simplesamlphp-1.10.0.orig/config/authsources.php simplesamlphp-1.10.0/config/authsources.php
--- simplesamlphp-1.10.0.orig/config/authsources.php	2012-09-25 12:16:10.000000000 +0200
+++ simplesamlphp-1.10.0/config/authsources.php	2012-09-30 19:35:19.000000000 +0200
@@ -22,7 +22,7 @@
 
 		// The entity ID of the IdP this should SP should contact.
 		// Can be NULL/unset, in which case the user will be shown a list of available IdPs.
-		'idp' => NULL,
+		'idp' => '{BASE_URL}/ssp/saml2/idp/metadata.php',
 
 		// The URL to the discovery service.
 		// Can be NULL/unset, in which case a builtin discovery service will be used.
@@ -49,6 +49,62 @@
 	),
 	*/
 
+    'my-userpass' => array(
+        'exampleauth:UserPass',
+
+        // Administrator               
+        'admin:adm1n' => array(
+            'uid' => array('admin'),
+            // entitlement to manage OAuth applications
+            'eduPersonEntitlement' => array('urn:vnd:oauth2:applications'),
+            'eduPersonPrincipalName' => array('admin@example.edu'),
+            'eduPersonAffiliation' => array('member', 'employee'),
+            'cn' => array('Carlos Catalano'),
+        ),
+
+        // Teacher
+        'teacher:teach3r' => array(
+            'uid' => array('teacher'),
+            // entitlement to view grades of all students and list of students
+            'eduPersonEntitlement' => array('urn:vnd:grades:administration'),
+            'eduPersonPrincipalName' => array('teacher@example.edu'),
+            'eduPersonAffiliation' => array('member', 'employee'),
+            'cn' => array('Margie Korn'),
+        ),
+
+        // Students
+        'jstroud:xDIosd9j' => array(
+            'uid' => array('jstroud'),
+            'eduPersonAffiliation' => array('member', 'student'),
+            'eduPersonPrincipalName' => array('jstroud@students.example.edu'),
+            'cn' => array('Jerry Stroud'),
+        ),
+        'mwisdom:LQ32xvaV' => array(
+            'uid' => array('mwisdom'),
+            'eduPersonAffiliation' => array('member', 'student'),
+            'eduPersonPrincipalName' => array('mwisdom@students.example.edu'),
+            'cn' => array('Myra Wisdom'),
+        ),
+        'bmcatee:ySVk4B9d' => array(
+            'uid' => array('bcatee'),
+            'eduPersonAffiliation' => array('member', 'student'),
+            'eduPersonPrincipalName' => array('bmcatee@students.example.edu'),
+            'cn' => array('Bobby Mcatee'),
+        ),
+        'jmatson:8r5yyIV3' => array(
+            'uid' => array('jmatson'),
+            'eduPersonAffiliation' => array('member', 'student'),
+            'eduPersonPrincipalName' => array('jmatson@students.example.edu'),
+            'cn' => array('Janie Matson'),
+        ),
+        'mcram:62xX1Fid' => array(
+            'uid' => array('mcram'),
+            'eduPersonAffiliation' => array('member', 'student'),
+            'eduPersonPrincipalName' => array('mcram@students.example.edu'),
+            'cn' => array('Margaret Cram'),
+        ),
+    ),
+
 	/*
 	'example-userpass' => array(
 		'exampleauth:UserPass',
diff -urN simplesamlphp-1.10.0.orig/config/config.php simplesamlphp-1.10.0/config/config.php
--- simplesamlphp-1.10.0.orig/config/config.php	2012-09-25 12:16:10.000000000 +0200
+++ simplesamlphp-1.10.0/config/config.php	2012-09-30 19:34:10.000000000 +0200
@@ -22,7 +22,7 @@
 	 * external url, no matter where you come from (direct access or via the
 	 * reverse proxy).
 	 */
-	'baseurlpath'           => 'simplesaml/',
+	'baseurlpath'           => '{BASE_URL}/ssp/',
 	'certdir'               => 'cert/',
 	'loggingdir'            => 'log/',
 	'datadir'               => 'data/',
@@ -69,7 +69,7 @@
 	 * metadata listing and diagnostics pages.
 	 * You can also put a hash here; run "bin/pwgen.php" to generate one.
 	 */
-	'auth.adminpassword'		=> '123',
+	'auth.adminpassword'		=> '{ADMIN_PASSWORD}',
 	'admin.protectindexpage'	=> false,
 	'admin.protectmetadata'		=> false,
 
@@ -81,7 +81,7 @@
 	 * A possible way to generate a random salt is by running the following command from a unix shell:
 	 * tr -c -d '0123456789abcdefghijklmnopqrstuvwxyz' </dev/urandom | dd bs=32 count=1 2>/dev/null;echo
 	 */
-	'secretsalt' => 'defaultsecretsalt',
+	'secretsalt' => '{SECRET_SALT}',
 	
 	/*
 	 * Some information about the technical persons running this installation.
@@ -98,7 +98,7 @@
 	 *
 	 * See this page for a list of valid timezones: http://php.net/manual/en/timezones.php
 	 */
-	'timezone' => NULL,
+	'timezone' => "Europe/Amsterdam",
 
 	/*
 	 * Logging.
@@ -171,7 +171,7 @@
 	 * one of the functionalities below, but in some cases you could run multiple functionalities.
 	 * In example when you are setting up a federation bridge.
 	 */
-	'enable.saml20-idp'		=> false,
+	'enable.saml20-idp'		=> true,
 	'enable.shib13-idp'		=> false,
 	'enable.adfs-idp'		=> false,
 	'enable.wsfed-sp'		=> false,
diff -urN simplesamlphp-1.10.0.orig/metadata/saml20-idp-hosted.php simplesamlphp-1.10.0/metadata/saml20-idp-hosted.php
--- simplesamlphp-1.10.0.orig/metadata/saml20-idp-hosted.php	2012-09-25 12:16:10.000000000 +0200
+++ simplesamlphp-1.10.0/metadata/saml20-idp-hosted.php	2012-09-30 19:34:10.000000000 +0200
@@ -14,14 +14,14 @@
 	'host' => '__DEFAULT__',
 
 	/* X.509 key and certificate. Relative to the cert directory. */
-	'privatekey' => 'server.pem',
-	'certificate' => 'server.crt',
+	'privatekey' => 'idp.pem',
+	'certificate' => 'idp.crt',
 
 	/*
 	 * Authentication source to use. Must be one that is configured in
 	 * 'config/authsources.php'.
 	 */
-	'auth' => 'example-userpass',
+	'auth' => 'my-userpass',
 
 	/* Uncomment the following to use the uri NameFormat on attributes. */
 	/*
diff -urN simplesamlphp-1.10.0.orig/metadata/saml20-idp-remote.php simplesamlphp-1.10.0/metadata/saml20-idp-remote.php
--- simplesamlphp-1.10.0.orig/metadata/saml20-idp-remote.php	2012-09-25 12:16:10.000000000 +0200
+++ simplesamlphp-1.10.0/metadata/saml20-idp-remote.php	2012-09-30 19:34:10.000000000 +0200
@@ -89,3 +89,12 @@
 	'SingleLogoutService'  => 'https://testidp.wayf.dk/saml2/idp/SingleLogoutService.php',
 	'certFingerprint'      => '04b3b08bce004c27458b3e85b125273e67ef062b'
 );
+
+$metadata['{BASE_URL}/ssp/saml2/idp/metadata.php'] = array(
+	'name' => array(
+		'en' => 'OAuth Demo IdP',
+	),
+	'SingleSignOnService' => '{BASE_URL}/ssp/saml2/idp/SSOService.php',
+	'SingleLogoutService' => '{BASE_URL}/ssp/saml2/idp/SingleLogoutService.php',
+	'certFingerprint' => '{CERT_FINGERPRINT}',
+);
diff -urN simplesamlphp-1.10.0.orig/metadata/saml20-sp-remote.php simplesamlphp-1.10.0/metadata/saml20-sp-remote.php
--- simplesamlphp-1.10.0.orig/metadata/saml20-sp-remote.php	2012-09-25 12:16:10.000000000 +0200
+++ simplesamlphp-1.10.0/metadata/saml20-sp-remote.php	2012-09-30 19:34:10.000000000 +0200
@@ -25,3 +25,9 @@
 	'simplesaml.nameidattribute' => 'uid',
 	'simplesaml.attributes' => FALSE,
 );
+
+$metadata['{BASE_URL}/ssp/module.php/saml/sp/metadata.php/default-sp'] = array(
+	'NameIDFormat' => 'urn:oasis:names:tc:SAML:2.0:nameid-format:persistent',
+	'AssertionConsumerService' => '{BASE_URL}/ssp/module.php/saml/sp/saml2-acs.php/default-sp',
+	'SingleLogoutService' => '{BASE_URL}/ssp/module.php/saml/sp/saml2-logout.php/default-sp',
+);
